jlm/llvm/opt/alias-analyses/PointsToGraph.cpp

```diff
@@ -161,6 +161,135 @@ PointsToGraph::AddImportNode(std::unique_ptr<PointsToGraph::ImportNode> node)
   return *tmp;
 }
 
+bool
+PointsToGraph::IsSupergraphOf(const jlm::llvm::aa::PointsToGraph & subgraph) const
+{
+  // Given a memory node representing a memory object in an RVSDG module, this function finds
+  // a memory node representing the same memory object in a different PointsToGraph.
+  // If no corresponding memory node exists in the graph, nullptr is returned
+  const auto GetCorrespondingMemoryNode =
+      [](const PointsToGraph::MemoryNode & node,
+         const PointsToGraph & graph) -> const PointsToGraph::MemoryNode *
+  {
+    if (auto allocaNode = dynamic_cast<const AllocaNode *>(&node))
+    {
+      if (auto it = graph.AllocaNodes_.find(&allocaNode->GetAllocaNode());
+          it != graph.AllocaNodes_.end())
+        return it->second.get();
+    }
+    else if (auto deltaNode = dynamic_cast<const DeltaNode *>(&node))
+    {
+      if (auto it = graph.DeltaNodes_.find(&deltaNode->GetDeltaNode());
+          it != graph.DeltaNodes_.end())
+        return it->second.get();
+    }
+    else if (auto importNode = dynamic_cast<const ImportNode *>(&node))
+    {
+      if (auto it = graph.ImportNodes_.find(&importNode->GetArgument());
+          it != graph.ImportNodes_.end())
+        return it->second.get();
+    }
+    else if (auto lambdaNode = dynamic_cast<const LambdaNode *>(&node))
+    {
+      if (auto it = graph.LambdaNodes_.find(&lambdaNode->GetLambdaNode());
+          it != graph.LambdaNodes_.end())
+        return it->second.get();
+    }
+    else if (auto mallocNode = dynamic_cast<const MallocNode *>(&node))
+    {
+      if (auto it = graph.MallocNodes_.find(&mallocNode->GetMallocNode());
+          it != graph.MallocNodes_.end())
+        return it->second.get();
+    }
+    else if (MemoryNode::Is<UnknownMemoryNode>(node))
+    {
+      return &graph.GetUnknownMemoryNode();
+    }
+    else if (MemoryNode::Is<ExternalMemoryNode>(node))
+    {
+      return &graph.GetExternalMemoryNode();
+    }
+    else
+      JLM_UNREACHABLE("Unknown type of MemoryNode");
+
+    return nullptr;
```

Comment:
This could just be `JLM_UNREACHABLE(...)`, no? Remove the else (and the return nullptr), and just put the UNREACHABLE at the end?

Reply:
No, all the other if bodies are supposed to return `nullptr` if they don't find a match. I can write `return nullptr` in each of them if it is clearer.

Reply:
You are absolutely right. I misread the code. Thanks for clarifying. LGTM.